src/neo/SmartContract/ApplicationEngine.Runtime.cs

```diff
@@ -183,8 +184,12 @@ protected internal ulong GetTime()
         /// <returns>The current script container.</returns>
         protected internal StackItem GetScriptContainer()
         {
-            if (ScriptContainer is not IInteroperable interop) throw new InvalidOperationException();
-            return interop.ToStackItem(ReferenceCounter);
+            return ScriptContainer switch
+            {
+                IInteroperable interop => interop.ToStackItem(ReferenceCounter),
+                null => null,
+                _ => throw new CatchableException("The script container cann't be converted to an IInteroperable object.")
```

Comment:
How can this ever happen if we don't have a bug in the node code?

Reply:
I don't think this exception should be catchable